python/dgl/nn/pytorch/explain/pgexplainer.py

```diff
@@ -0,0 +1,363 @@
+            contribution of the edge.
+
+        Examples
+        --------
+
+        >>> import torch as th
+        >>> import torch.nn as nn
+        >>> import dgl
+        >>> from dgl.data import GINDataset
+        >>> from dgl.dataloading import GraphDataLoader
+        >>> from dgl.nn import GraphConv, PGExplainer
+        >>> import numpy as np
+
+        >>> # Define the model
+        >>> class Model(nn.Module):
+        ...     def __init__(self, in_feats, out_feats):
+        ...         super().__init__()
+        ...         self.conv = GraphConv(in_feats, out_feats)
+        ...         self.fc = nn.Linear(out_feats, out_feats)
+        ...         nn.init.xavier_uniform_(self.fc.weight)
+        ...
+        ...     def forward(self, g, h, embed=False, edge_weight=None):
+        ...         h = self.conv(g, h, edge_weight=edge_weight)
+        ...         if not embed:
+        ...             g.ndata['h'] = h
+        ...             hg = dgl.mean_nodes(g, 'h')
+        ...             return self.fc(hg)
+        ...         else:
+        ...             return h
+
+        >>> # Load dataset
+        >>> data = GINDataset('MUTAG', self_loop=True)
+        >>> dataloader = GraphDataLoader(data, batch_size=64, shuffle=True)
+
+        >>> # Train the model
+        >>> feat_size = data[0][0].ndata['attr'].shape[1]
+        >>> model = Model(feat_size, data.gclasses)
+        >>> criterion = nn.CrossEntropyLoss()
+        >>> optimizer = th.optim.Adam(model.parameters(), lr=1e-2)
+        >>> for bg, labels in dataloader:
+        ...     preds = model(bg, bg.ndata['attr'])
+        ...     loss = criterion(preds, labels)
+        ...     optimizer.zero_grad()
+        ...     loss.backward()
+        ...     optimizer.step()
+
+        >>> # Initialize the explainer
+        >>> explainer = PGExplainer(model, data.gclasses)
+
+        >>> # Train the explainer
+        >>> # Define explainer temperature parameter
+        >>> init_tmp, final_tmp = 5.0, 1.0
+        >>> optimizer_exp = th.optim.Adam(explainer.parameters(), lr=0.01)
+        >>> for epoch in range(20):
+        ...     tmp = float(init_tmp * np.power(final_tmp / init_tmp, epoch / 20))
+        ...     for bg, labels in dataloader:
+        ...          loss = explainer.train_step(bg, bg.ndata['attr'], tmp)
+        ...          optimizer_exp.zero_grad()
+        ...          loss.backward()
+        ...          optimizer_exp.step()
+
+        >>> # Explain the prediction for graph 0
+        >>> graph, l = data[0]
+        >>> graph_feat = graph.ndata.pop("attr")
+        >>> probs, edge_weight = explainer.explain_graph(graph, graph_feat)
+        """
+        self.model = self.model.to(graph.device)
+        self.elayers = self.elayers.to(graph.device)
+
+        embed = self.model(graph, feat, embed=True, **kwargs)
+        embed = embed.data
+
+        edge_idx = graph.edges()
+
+        row, col = edge_idx
```

Comment:
```suggestion
        col, row = edge_idx
```
Equivalently, you have `src, dst = edge_idx`. L350 is for getting the reverse edges rather than the original edges.

Reply:
Fixed.